language/diem-tools/writeset-transaction-generator/src/main.rs

```diff
@@ -47,6 +51,50 @@ enum Command {
         /// Blockchain height
         version: u64,
     },
+    /// Verify if a blob is generated by the checked-in artifact.
+    #[structopt(name = "verify-release")]
+    VerifyDiemFrameworkRelease {
+        /// Path to the serialized bytes of WriteSet.
+        #[structopt(parse(from_os_str))]
+        writeset_path: PathBuf,
+        /// True if the serialized byte has the type of WriteSetPayload.
+        #[structopt(long)]
+        is_payload: bool,
+    },
+}
+
+/// Checkin the release artifact to a local file to make sure the release process is reproducible
+#[derive(Serialize, Deserialize)]
+struct ReleaseArtifact {
+    remote_json_rpc_url: String,
```

Comment:
I think it would be more future-proof to use a ChainID instead of URL (both here and in the inputs to the CLI). The problem: the URLs are not stable identifiers (e.g., they have changed several times already due to the Diem renaming), whereas a chain ID is a stable identifier by definition.